packages/ipfs-core-types/src/files/index.d.ts

```diff
@@ -0,0 +1,355 @@
+import { AbortOptions, IPFSPath } from '../basic'
+import { ToMTime } from './files'
+import CID, { CIDVersion } from 'cids'
+import { CodecName } from 'multicodec'
+import { HashName } from 'multihashes'
+import { Mtime } from 'ipfs-unixfs'
+import type { AddProgressFn } from '../root'
+
+export interface API<OptionExtension = {}> {
+  chmod: (path: string, mode: number | string, options?: ChmodOptions & OptionExtension) => Promise<void>
+  cp: (from: IPFSPath | IPFSPath[], to: string, options?: CpOptions & OptionExtension) => Promise<void>
```

Comment:
whats a `IPFSPath` ? `ipfs://<cid>`, `<cid>/filename.png` ?

Reply:
Variously we have:

`Qmfoo`
`/ipfs/Qmfoo`
`/ipfs/Qmfoo/some/deep/file.txt`
`/ipns/Qmfoo` (!)

Here it could also be:

`/path/in/my/mfs.txt`

I'm starting to be of the opinion that these things should just be `CID`s or MFS paths.  If you have `/ipfs/Qmfoo/some/deep/file.txt` you should `ipfs.dag.resolve` it first and get a `CID`.